lib/blocks/registry.py

```
@@ -33,23 +59,64 @@ def _discover_blocks(self) -> None:
+                    )
+                    for _name, obj in inspect.getmembers(module, inspect.isclass):
+                        if (
+                            issubclass(obj, BaseBlock)
+                            and obj not in (BaseBlock, BaseMultiplierBlock)
+                            and obj.__module__ == module.__name__
                         ):
-                            self._blocks[obj.__name__] = obj
+                            entries[obj.__name__] = BlockEntry(
+                                block_class=obj, source=source
+                            )
                 except Exception as e:
```

Ruff (0.15.4)

[warning] Do not catch blind exception: `Exception`

(BLE001)